### PR TITLE
Fix eightwone bug

### PR DESCRIPTION
**Pull request by kschluns**
TL;DR - Found a bug in `eightwone.com` where certain product names have typos which was resulting in zero associated vulnerabilities for `Exchange Server 2019 CU7`.

Solution: Added manual correction to `automation/parse_exchange_versions.py` and re-ran the automation pipeline.

---------------------------------------------------------------------------------------------------------------------------

**LONGER STORY:**

I was validating some random Exchange CVEs to confirm the data quality and came across an issue with `CVE-2020-17117` where I'm not seeing it on any of the Exchange Server 2019 CU7 builds. In fact, none of these build have any vulnerabilities as of today.

Source: https://github.com/righel/ms-exchange-version-nse/blob/7738ce14abc1d7a9b318f1ac8ce1ca371be2470d/ms-exchange-versions-cves-dict.json#L15406-L15429

On my local laptop, I ran the four scripts specified in the `.github/workflows/main.yml` file:
https://github.com/righel/ms-exchange-version-nse/blob/7738ce14abc1d7a9b318f1ac8ce1ca371be2470d/.github/workflows/main.yml#L21-L24

After running the third script, I was surprised because the problem appeared fixed in the `ms-exchange-versions-cves-dict` file. However, when I ran the fourth script, the problem appeared again and all the CVEs were gone for 2019CU7. 

I was able to trace the problem back to an error in this file, which incorrectly maps productId `11792` to build number `15.1.2044` when it should map to build `15.2.721`.
https://github.com/righel/ms-exchange-version-nse/blob/7738ce14abc1d7a9b318f1ac8ce1ca371be2470d/product_id_build_map.json#L50

Tracing this down even further, this is caused by an error in one of your build number sources (`eightwone.com`). The source accidentally marked the red-highlighted row as 2019CU7 when it should be 2019CU17.
Screenshot 2024-03-15 at 1 50 26 PM

The same issue is also present in `eightwone.com` for build 15.1.1713, which should only map to 2016CU12 but it is also mapping to 2016CU1 and 2016CU13.

### CODE REVIEW

**Comment by righel** [comment]
Awesome! Good findings and thanks for the deep dive.
Have you reported this to https://eightwone.com? there is a contact form and he fixed some similar error I found in the past.

**Comment by kschluns** [comment]
Happy to help! This is awesome work. I've been struggling with this problem for a while, so your work is much appreciated. 

And thanks for the suggestion, will reach out to them.
